tornado/gen.py

```diff
@@ -344,7 +345,7 @@ def wrapper(*args, **kwargs):
                     # attribute on the Future.
                     # (Github issues #1769, #2229).
                     runner = Runner(result, future, yielded)
-                    future.add_done_callback(lambda _: runner)
+                    future.add_done_callback(runner)
```

Comment:
Is it not a problem that the `lambda returns the runner object` and the new `runner.__call__ returns None`?

Reply:
Nope, the return value serves no purpose. The lambda is there only to hold a reference to the runner, see the comment above the original line.